File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> st = new Stack<>();
        String[] s = path.split("/");
        for(int i=0;i<s.length;i++){
            if(s[i].isEmpty()) continue;
            if("..".equals(s[i])){
                if(!st.isEmpty())
                    st.pop();
            }
            else if(".".equals(s[i]))
                continue;
            else
                st.push(s[i]);
        }
        String res = "";
        if(st.isEmpty())
            return "/";
        while(!st.isEmpty()){
            res = "/"+st.pop() +res;
        }
        return res;        
    }
}

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        arr = path.split('/')
        st = []
        for ele in arr:
            if ele == '/' or ele == '.' or ele == '':
                continue
            if ele == '..' :
                if len(st) > 0:
                    st.pop()
            else:
                st.append(ele)
        res = ''
        if len(st) == 0:
            return '/'
        for i in range(len(st)):
            res += f'/{st[i]}'
        return res
                
        